codespan-reporting/src/term/renderer.rs

```diff
@@ -240,6 +240,20 @@ impl<'writer, 'config> Renderer<'writer, 'config> {
         // FIXME: Use the number of trimmed placeholders when rendering single line carets
         let source = source.trim_end_matches(['\n', '\r', '\0'].as_ref());
 
+        // get the byte index of the first non-whitespace character
+        let text_start = source.find(|c| !char::is_whitespace(c)).unwrap_or(0);
+
+        // get the byte index of the first trailing whitespace character
+        let text_end = source
+            .char_indices()
+            .rev()
+            .take_while(|(_, c)| c.is_whitespace())
+            .last()
+            .map(|(index, _)| index)
+            .unwrap_or_else(|| source.len());
```

Comment:
This could also be rewritten using str::rfind, but because we have to find the first trailing non-whitespace character, we would have to find the char lengt of the character to adjust, which makes it not much clearer than the curren solution.
```suggestion
        let text_end = source
            .rfind(|c| !char::is_whitespace(c))
            .map(|index|
                index + source[index..]
                    .chars()
                    .next()
                    .map_or(0, |c| c.len_utf8())
            )
            .unwrap_or_else(|| source.len());
```